app/webpacker/components/CompetitionsOverview/CompetitionsView.js

```diff
@@ -68,12 +68,41 @@ function CompetitionsView() {
     },
   });
 
-  const competitions = rawCompetitionData?.pages.flatMap((page) => page.data)
+  const baseCompetitions = rawCompetitionData?.pages.flatMap((page) => page.data)
     .filter((comp) => (
       (!isCancelled(comp) || debouncedFilterState.shouldIncludeCancelled)
       && (debouncedFilterState.selectedEvents.every((event) => comp.event_ids.includes(event)))
     ));
 
+  const compIds = baseCompetitions?.map((comp) => comp.id) || [];
+
+  const {
+    data: compRegistrationData,
+    isFetching: regDataIsPending,
+  } = useQuery({
+    queryFn: () => fetchJsonOrError(apiV0Urls.competitions.registrationData, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+      body: JSON.stringify({ ids: compIds }),
+    }),
+    queryKey: ['registration-info', ...compIds],
+    enabled: shouldShowRegStatus,
+    // This is where the magic happens: Using `keepPreviousData` makes it so that
+    //   all previously loaded indicators are held in-cache while the fetcher for the next
+    //   batch is running in the background. (Adding comment here because it's not in the docs)
+    placeholderData: keepPreviousData,
+    select: (data) => data.data,
```

Comment:
Do we have to have `data.data`? Does one of them have a better name?

Reply:
This pattern is a common problem with using your own fetcher methods in Tanstack query. It's even present in their own documentation and other files already suffer from it. We can think about a solution once we've made the full switch.